protocol/storage/types.go

```diff
@@ -60,8 +61,24 @@ type DealProposal struct {
 	// will use to pay the miner. It should be verifiable by the
 	// miner using on-chain information.
 	Payment PaymentInfo
+}
+
+// Unmarshal a DealProposal from the given bytes.
+func (dp *DealProposal) Unmarshal(b []byte) error {
+	return cbor.DecodeInto(b, dp)
+}
 
-	// Signature types.Signature
+// Marshal the DealProposal into bytes.
+func (dp *DealProposal) Marshal() ([]byte, error) {
+	return cbor.DumpObject(dp)
+}
+
+// SignedDealProposal is the information sent over the wire, when a clinet proposes a deal to a miner.
+// The SignedDealProposal contains the DealProposal and a Signature for the proposal.
```

Comment:
Nit: this comment is mostly redundant given the code. Suggest "SignedDealProposal is a deal proposal signed by the proposing client".